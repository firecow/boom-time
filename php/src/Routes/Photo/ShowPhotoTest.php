<?
declare(strict_types=1);

namespace App\Routes\Photo;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class ShowPhotoTest extends RegressionTest
{

    /**
     * @return string
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/photo/5b3e0bc586463d34800048a3");
        return $routeResponse->prettyPrint();
    }
}

