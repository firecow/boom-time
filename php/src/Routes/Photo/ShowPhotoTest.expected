HTTP/1.1 200 Ok
Content-Length: 849
Content-Type: image/png
Expires: Thu, 22 Feb 2018 13:17:02 GMT

Image MD5: 5ac2405d8d360e8345ef0f9ea14e49c8

- 		"$date": "2018-02-16T13:02:02.000Z"
+ 		"$date": "2018-02-22T13:02:02.000Z"
- 	"seenCount": 5,
+ 	"seenCount": 6,
